File: UI/iOS/iChatLogin/Data/MockAuthService.swift
import Foundation

enum AuthResult: Equatable {
    case requestSuccess
    case verifySuccess(token: String, isNewUser: Bool)
    case invalidCode
    case networkError
}

/// Mock 实现：本地模拟延迟 + 固定通过码 "123456"。仅用于第一期 demo 与测试。
final class MockAuthService: AuthService {
    func requestCode(phone: String) async throws -> AuthResult {
        try await Task.sleep(nanoseconds: 800_000_000)
        return .requestSuccess
    }

    func verifyCode(phone: String, code: String) async throws -> AuthResult {
        try await Task.sleep(nanoseconds: 600_000_000)
        return code == "123456"
            ? .verifySuccess(token: "mock-token", isNewUser: false)
            : .invalidCode
    }
}

File: UI/Android/app/src/main/kotlin/com/ichat/login/data/MockAuthService.kt
package com.ichat.login.data

import kotlinx.coroutines.delay

sealed class AuthResult {
    data object RequestSuccess : AuthResult()
    data class VerifySuccess(val token: String, val isNewUser: Boolean) : AuthResult()
    data object InvalidCode : AuthResult()
    data object NetworkError : AuthResult()
}

/**
 * Mock 实现：本地模拟延迟 + 固定通过码 "123456"。仅用于第一期 demo 与测试。
 */
class MockAuthService : AuthService {
    override suspend fun requestCode(phone: String): AuthResult {
        delay(800)
        return AuthResult.RequestSuccess
    }

    override suspend fun verifyCode(phone: String, code: String): AuthResult {
        delay(600)
        return if (code == "123456") AuthResult.VerifySuccess(token = "mock-token", isNewUser = false)
        else AuthResult.InvalidCode
    }
}
